### DISCUSSION TITLE
Changing Compression Level

### DISCUSSION THREAD

**Discussion by Deedasmi** [rustic-rs/rustic#642] [Q&A] [answered]
Apologies for making an issue, I forgot discussions were a thing.

I changed the compression level of my repository, but there is no change to the backup process. Tried to backup, forget, prune, backup force, and snapshots never change in size. Is there something I need to do to repack everything?

**Comment by aawsome** [comment, marked as answer]
Yes, if you change the compression level, only newly generated packs are affected but not existing ones.

If you have uncompressed packs, you can compress them with `rustic prune --repack-uncompressed`  but if you have already compressed packs and only want to change the compression level, you have to repack all packs (rustic only know IF a pack is compressed, but not the actual compression level).

To repack all packs, use `rustic prune --repack-all` - make sure to NOT specify `--repack-fast`, though!
If you can ensure that no other process is accessing the repository in parallel, you can add the option `--instant-delete`.
